bleak/backends/bluezdbus/client.py

```diff
@@ -148,7 +149,7 @@ async def connect(self, dangerous_use_bleak_cache: bool = False, **kwargs) -> bo
 
         manager = await get_global_bluez_manager()
 
-        async with async_timeout(timeout):
+        async with async_timeout(timeout), AsyncExitStack() as stack:
```

Comment:
Just noticed an issue too late. We need a new stack for each while loop, otherwise we would end up leaking on retries. I'll fix it.